src/middlewared/middlewared/api/v25_04_0/smb.py

```diff
@@ -63,3 +99,65 @@ class GetSmbAclArgs(BaseModel):
 
 class GetSmbAclResult(SetSmbAclResult):
     pass
+
+
+SMBEncryption = Literal['DEFAULT', 'NEGOTIATE', 'DESIRED', 'REQUIRED']
+
+
+class SmbServiceEntry(BaseModel):
+    id: int
+    netbiosname: NetbiosName
```

Comment:
If `netbiosalias` (below) does not have the 15 char length restriction, then maybe...
    `netbiosname: Annotated[NetbiosName, Field(max_length=15)]`
else use the suggested change in `base/string.py`

Reply:
I changed validator itself.